utils.js

```diff
@@ -0,0 +1,140 @@
+    $('link[href$="site.webmanifest"]').remove();
+    // lets get rid of all the head tags since we are populating them with the SEO component
+    $('meta[content*="{{"]').remove();
+    //
+    // padd as per https://stackoverflow.com/questions/11124777/twitter-bootstrap-navbar-fixed-top-overlapping-site
+    $('head').append('<style>{`body { padding-top: 40px; } @media screen and (max-width: 768px) { body { padding-top: 0px; } } `}</style>');
+    $('head').append('<style>{`#grid-box { position: relative } `}</style>');
+    $('head').append('<style>{`html { min-height:100%; position: relative; }`}</style>');
+
+    $('.nav-link[href="https://plugins.jenkins.io/"]').attr('href', '/');
+    $('#grid-box').append('{children}');
+    $('#creativecommons').append('<SiteVersion />');
+
+    const keyConversion = {
+        class: 'className',
+        'charSet': 'charset',
+        'http-equiv': 'httpEquiv',
+        'stop-color': 'stopColor',
+        'crossorigin': 'crossOrigin'
+    };
+
+    const handleNode = (node, indent = 0) => {
+        
+        const prefix = ''.padStart(6+indent);
+        if (node.name === 'link' && node.attribs && node.attribs.rel === 'stylesheet') {
+            delete node.attribs.crossorigin;
+            node.attribs.type = 'text/css';
+            node.attribs.media = 'all';
+        }
+        const attrs = Object.entries(node.attribs || {}).map(([key, val]) => {
+            key = keyConversion[key] || key;
+            val = val.replace(/"/g, '\\"');
+            return `${key}="${val}"`;
+        }).join(' ');
+        if (node.name === 'script') {
+            // FIXME - handle me
+            const text = node.children.map(child => {
+                if (child.type === 'text') {
+                    return child.data;
+                }
+                throw new Error(`not sure how to handle ${child.type}`);
+            });
+            lines.push(`${prefix}<${node.name} ${attrs} dangerouslySetInnerHTML={{__html: ${JSON.stringify(text)}}} />`);
+            return;
+        } else if (node.type === 'comment') {
+            return;
+        } else if (node.type === 'text') {
+            const text = node.data;
+            lines.push(`${prefix}${text}`);
+        } else if (node.children && node.children.length) {
+            lines.push(`${prefix}<${node.name} ${attrs}>`);
+            node.children.forEach(child => handleNode(child, indent+2));
+            lines.push(`${prefix}</${node.name}>`);
+        } else {
+            if (!node.name) {
+                console.log(node);
+            }
+            if (node.name === 'siteversion') {
+                node.name = 'SiteVersion';
+            }
+            lines.push(`${prefix}<${node.name} ${attrs} />`);
+        }
+    };
+
+    lines.push('export default function Layout({ children, id }) {');
```

Comment:
couldn't we just add all this with a multi line string?

Reply:
Probably, but i like consistency?